common/txmgr/types/storage_service.go

```diff
@@ -0,0 +1,95 @@
+package types
+
+import (
+	"context"
+	"time"
+
+	uuid "github.com/satori/go.uuid"
+)
+
+// NEWTX, TX, TXATTEMPT will be converted from generic types to structs at a future date to enforce design and type checks
+type TxStorageService[ADDR any, CHAINID any, HASH any, NEWTX any, R any, TX any, TXATTEMPT any, TXID any, TXMETA any] interface {
```

Comment:
i wonder if there's a way to to avoid using `any` for `ADDR`, `CHAINID`, and `HASH`

either a type that would work more commonly like `[]byte` or something where defining an underlying interface the type must conform to 🤔 

but that would probably be better in a separate PR

Reply:
@jmank88 thoughts on the use of generic types here?

Reply:
Yeah I wouldn't be opposed to using []byte or string - we already have that UX elsewhere.

Reply:
FWIW we could still get a bit of flexibility with `[]byte`/`string` by using custom types that have that underlying type but also extend it with custom methods. Then we can still use code that works on the plain types, or use `~` for simpler generics:
> The expression ~string means the set of all types whose underlying type is string. This includes the type string itself as well as all types declared with definitions such as type MyString string.

Reply:
+1 to `~[]byte` or `~string`

Reply:
Lets tackle this in new PRs, but I feel an interface is better than above approaches. We can dwell deeper in those PRs, about what approach is better.

Reply:
Prefer interface to `any`